practice-app/backend/api/views/poll_views.py

```diff
@@ -0,0 +1,79 @@
+import json
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from api.serializers import poll_serializer
+import os
+from dotenv import load_dotenv
+from api.models.poll_model import Poll
+import requests
+
+# Create your views here.
+
+
+class createPoll(APIView):
+    serializer_class = poll_serializer
+    def post(self,request, format = None):
+        serializer = self.serializer_class(data=request.query_params)
+        try:
+            if serializer.is_valid():
+                load_dotenv()
+                Poll_SECRET_KEY = os.getenv('Poll_SECRET_KEY')
+                url = 'https://api.pollsapi.com/v1/create/poll'
+                headers = {'Content-Type': 'application/json'}
+                headers.update({'api-key': Poll_SECRET_KEY})
+                question_element = serializer.data.get('question')
+                optionsList = []
+                firstOption_element = serializer.data.get('firstOption')
+                secondOption_element = serializer.data.get('secondOption')
+                thirdOption_element = serializer.data.get('thirdOption')
+                fourthOption_element = serializer.data.get('fourthOption')
+                optionsList.append(firstOption_element)
+                optionsList.append(secondOption_element)
+                optionsList.append(thirdOption_element)
+                optionsList.append(fourthOption_element)
```

Comment:
Fena tekrar var buralarda bu örnek için demiyorum ama bunları fonksiyon yaparsan daha hoş ve test edilebilir olur.